packages/@react-stately/tag/src/useTagGroupState.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {GridCollection, GridState, useGridState} from '@react-stately/grid';
+import {ITagGroupCollection, TagGroupProps} from '@react-types/tag';
+import {useEffect, useMemo, useRef} from 'react';
+import {useListState} from 'react-stately';
+
+export interface TagGroupState<T>  extends GridState<T, ITagGroupCollection<T>>{}
+
+/**
+ * Provides state management for a TagGroup component.
+ */
+export function useTagGroupState<T extends object>(props: TagGroupProps<T>): TagGroupState<T> {
+  let listState = useListState(props);
+
+  let gridCollection = useMemo(() => new GridCollection({
+    columnCount: 1, // unused, but required for grid collections
+    items: [...listState.collection].map(item => ({
+      type: 'item',
+      childNodes: [{
+        ...item,
+        index: 0,
+        type: 'cell'
+      }]
+    }))
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }), [listState.collection, props.allowsRemoving]);
+
+  let state = useGridState({
+    ...props,
+    collection: gridCollection,
+    focusMode: 'row'
+  });
+
+  let keyToRestoreOnRemove = useRef(null);
+
+  useEffect(() => {
+    // If the focused key is removed, restore focus to the tag before, or tag after if no tag before.
+    if (!listState.collection.getItem(state.selectionManager.focusedKey)) {
+      state.selectionManager.setFocusedKey(keyToRestoreOnRemove.current);
+    }
+    keyToRestoreOnRemove.current = listState.collection.getKeyBefore(state.selectionManager.focusedKey) || listState.collection.getKeyAfter(state.selectionManager.focusedKey);
+  }, [listState.collection, state.collection, state.selectionManager]);
```

Comment:
Also since it seemed like focus was already making it onto another tag, I'm surprised that useSelectableItem wasn't updating the selectionManager's focused key via https://github.com/adobe/react-spectrum/blob/b35d5c02fe900badccd0cf1a8f23bb593419f238/packages/%40react-aria/selection/src/useSelectableItem.ts#L158-L161. Perhaps that doesn't fire/isn't supposed to fire, didn't dig too deeply.

Reply:
I think the focused key was being set, but the key was 'row-N' instead of 'N'. I'll take another look.

Reply:
If that is the case, that sounds similar to an issue we used to have in ListView when we were still using useGridRow + useGridCell for ListView if I recall correctly (I think it was the due to having the grid collection and the list collection but they contained different bits of information). 

Thinking out loud, maybe we could use useGridList and useGridListItem here instead and tear out the keydown handler from useGridList so that the TagKeyboadDelegate still works? No need to implement/change anything here, I haven't thought this one through just yet

Reply:
> maybe we could use useGridList and useGridListItem here instead

This works, except for useGridListItem keydown taking Left/Right arrow keys. Maybe I can just update useTag to look similar to useGridListItem but remove that keydown handling and anything else not needed.

Reply:
Bleh, was deleting the keydown handler returned from useGridListItem insufficient? If that's the case perhaps evaluate what stuff would need to be moved into useTag from useGridListItem and we can see what approach is easiest/appropriate.

Reply:
Ahh, yeah that does work. I thought I tried that, but I probably had something missing elsewhere. I'll go forward with trying this.

Reply:
Updated to use `useGridList`. Should be working how we now. Also nows focuses the next item after deletion instead of previous, which seems more natural.